Matchismo/Matchismo/View/PlayingCardView.mm

```diff
@@ -0,0 +1,213 @@
+//
+//  PlayingCardView.m
+//  Matchismo
+//
+//  Created by Otto Olkkonen on 20/06/2022.
+//
+
+#import "PlayingCardView.h"
+
+#import "CardViewConstants.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface PlayingCardView()
+@property (nonatomic) CGFloat faceCardScaleFactor;
+@end
+
+@implementation PlayingCardView
+
+@synthesize faceCardScaleFactor = _faceCardScaleFactor;
+
+- (CGFloat)cornerOffset {
+  return [self cornerRadius] / 3.0;
+}
+
+- (CGFloat)faceCardScaleFactor {
+  if (!_faceCardScaleFactor) {
+    _faceCardScaleFactor = DEFAULT_FACE_SCALE_FACTOR;
+  }
+  return  _faceCardScaleFactor;
+}
+
+- (void)setFaceCardScaleFactor:(CGFloat)faceCardScaleFactor {
+  _faceCardScaleFactor = faceCardScaleFactor;
+  [self setNeedsDisplay];
+}
+
+- (void)setRank:(NSUInteger)rank {
+  _rank = rank;
+  [self setNeedsDisplay];
+}
+
+- (void)setSuit:(NSString *)suit {
+  _suit = suit;
+  [self setNeedsDisplay];
+}
+
+- (void)setFaceUp:(BOOL)faceUp {
+  _faceUp = faceUp;
+  [self setNeedsDisplay];
+}
+
+- (NSString *)rankAsString {
+  return @[@"?", @"A", @"2", @"3", @"4", @"5", @"6", @"7", @"8", @"9", @"10", @"J", @"Q", @"K"]
+          [self.rank];
+}
+
+- (UIColor *)textColor {
+  return ([self.suit isEqualToString:@"♥︎"] || [self.suit isEqualToString:@"♦︎"])
+           ? UIColor.redColor : UIColor.blackColor;
+}
+
+- (void)drawRect:(CGRect)rect {
```

Comment:
Is this something that the exercise guided you into doing?
If not, just FYI that it's generally preferable to make UI components by constructing view hierarchies rather than "drawing" stuff with with CGContext or whatever in `drawRect`.  Not just because of performance (using `drawRect` pulls rendering over to the CPU that would otherwise have been done on the GPU) but also because it makes the components much easier to work with.